ConsoleApp/src/DemoArgProcessing/Options/Hidden/OptServerInfo.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using ConsoleApp.GenericArgProcessing;
+using DemoParser.Parser;
+using DemoParser.Parser.Components.Messages;
+using DemoParser.Parser.Components.Packets;
+using DemoParser.Utils;
+
+namespace ConsoleApp.DemoArgProcessing.Options.Hidden
+{
+
+    public class OptServerInfo : DemoOption<OptServerInfo.InfoFlags>
+    {
+
+        public static readonly ImmutableArray<string> DefaultAliases = new[] { "--server-info" }.ToImmutableArray();
+
+        private const int FmtIdt = -25;
+
+        [Flags]
+        public enum InfoFlags
+        {
+            ServerCountOnly = 1,
+        }
+
+        public OptServerInfo() : base(
+            DefaultAliases,
+            Arity.ZeroOrOne,
+            "Print SvcServerInfo" +
+            $"\nUse flag \"{InfoFlags.ServerCountOnly}\" or \"1\" to print server count only.",
+            "flags",
+            Utils.ParseEnum<InfoFlags>,
+            default,
+            true)
+        { }
+
+
+        protected override void AfterParse(DemoParsingSetupInfo setupObj, InfoFlags arg, bool isDefault)
+        {
+            setupObj.ExecutableOptions++;
+        }
+
+        protected override void Process(DemoParsingInfo infoObj, InfoFlags arg, bool isDefault)
+        {
+            infoObj.PrintOptionMessage("searching for SvcServerInfo");
+            try
+            {
+                bool any = false;
+                foreach (SvcServerInfo info in GetServerInfo(infoObj.CurrentDemo))
+                {
+                    any = true;
+                    WriteServerInfo(info, Console.Out, arg == InfoFlags.ServerCountOnly);
+                    break;
+                }
+                if (!any)
+                    Utils.Warning("SvcServerInfo not found");
+            }
+            catch (Exception)
+            {
+                Utils.Warning("Search for SvcServerInfo failed.\n");
+            }
+        }
+
+        public static IEnumerable<SvcServerInfo> GetServerInfo(SourceDemo demo)
+        {
+            return
+                from packet in demo.FilterForPacket<Packet>()
+                from message in packet.MessageStream
+                where message?.GetType() == typeof(SvcServerInfo)
+                select (SvcServerInfo)message;
+        }
+
+        public static void WriteServerInfo(SvcServerInfo info, TextWriter tw, bool server_count_only = false)

```

Comment:
If the arg to the option is a flag, then I think you should change `bool server_count_only` to `InfoFlags flags`.